agent/http.go

```diff
@@ -432,12 +432,20 @@ func (s *HTTPHandlers) wrap(handler endpoint, methods []string) http.HandlerFunc
 		}
 
 		handleErr := func(err error) {
-			httpLogger.Error("Request error",
-				"method", req.Method,
-				"url", logURL,
-				"from", req.RemoteAddr,
-				"error", err,
-			)
+			if req.Context().Err() != nil {
```

Comment:
Hmm, actually not sure about this. I wasn't thinking about the fact that this is the `req.Context` directly rarthe than a test against any `err` that might be returned by the handler.

In that case I don't know of any way the request context would error with `DeadlineExceeded` so this is probably equivalent, but what do you think about being explicit with the test given that this warning would be inappropriate if there ever was a way for http requests to inherently timeout?

Reply:
Because of the context hierarchy, I think the only way for `req.Context().Err()` to be not nil is if either:
* the request itself was cancelled by the user making the request
* if [http.Server.BaseContext](https://pkg.go.dev/net/http#Server) is set, and something cancels that context

I don't think we use `BaseContext` yet, but even if we did I think this message would be appropriate. Generally the only reason for the BaseContext to be cancelled would be if the server was shutting down, in which case logging the request as cancelled would also be accurate.

Reply:
Thinking about it more, I guess because `BaseContext` is a function, that could act as the middleware, and it could set a deadline, so I think you're right it is possible. 

I guess in most cases the `err` here would include the context about it being a timeout or a cancel, so maybe this log line will be ok in that case as well? I think it's a bit hard to say until we start using `BaseContext`.